hs_collection_resource/page_processors.py

```diff
@@ -0,0 +1,75 @@
+from django.db.models import Q
+from mezzanine.pages.page_processors import processor_for
+from crispy_forms.layout import Layout, HTML
+
+from hs_core import page_processors
+from hs_core.views import add_generic_context
+from hs_core.models import BaseResource
+from hs_access_control.models import PrivilegeCodes
+
+from .models import CollectionResource
+
+@processor_for(CollectionResource)
+def landing_page(request, page):
+    content_model = page.get_content_model()
+    edit_resource = page_processors.check_resource_mode(request)
+
+    user = request.user
+    if user.is_authenticated():
+        # get a list of resources with effective OWNER privilege
+        owned_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.OWNER)
+        # get a list of resources with effective CHANGE privilege
+        editable_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.CHANGE)
+        # get a list of resources with effective VIEW privilege
+        viewable_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.VIEW)
+
+        owned_resources = list(owned_resources)
+        editable_resources = list(editable_resources)
+        viewable_resources = list(viewable_resources)
+        discovered_resources = list(user.ulabels.my_resources)
+
+        user_all_accessible_resource_list = (owned_resources + editable_resources + \
+                                             viewable_resources + discovered_resources)
+    else: # anonymous user
+        user_all_accessible_resource_list = list(BaseResource.objects. \
```

Comment:
@pkdash does BaseResource.discoverable_resources.all() return all discoverable and public resources?